### PR TITLE
test : unit & e2e tests for whisp subscription and PUB_SUB Module

### PR DESCRIPTION
**Pull request by EL-HARBYLY-Abderrahman**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Design and implement a new unit & e2e tests for the whisp subscription (PUB_SUB)
- 100% coverage of the subscription code
- Fix tests and code that fell in warning issues
- Update the tests section of the documentation

* **What is the current behavior?** (You can also link to an open issue here)
- Today, the subscription service (PUBSUB) is used when creating whisps 
- There is no test (unit/e2e/integration) for this section

* **What is the new behavior (if this is a feature change)?**
- Implement a new unit & e2e tests for the whisp subscription (PUB_SUB)
- Fix tests and code that fell in warning issues

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No (dev dependency and unit & e2e tests changes only)

* **Other information**:

### CODE REVIEW

**Comment by kiru42** [comment]
@EL-HARBYLY-Abderrahman seems like the tests job was cancelled with this following error : 

`Ran all test suites.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
Error: The operation was canceled.`

Is it normal?

**Comment by EL-HARBYLY-Abderrahman** [comment]
> @EL-HARBYLY-Abderrahman seems like the tests job was cancelled with this following error :
> 
> ```
> Ran all test suites.
> Jest did not exit one second after the test run has completed.
> 
> This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
> Error: The operation was canceled.
> ```
> 
> Is it normal?

No, isn't normal. 
This is the error in I mentioned in the daily meeting. I will ask @MeStrak about it.

File: docs/testing.md [hunk 1]
```diff
@@ -11,7 +11,22 @@ $ npm run test:e2e
...
 
-## End to End Tests
+## Unit Tests
+The unit test is a way of testing a unit , the smallest possible unit of code that can be logically isolated in a system. These units are tested to determine if there are any issues 
+when the developer make the change in the code base.
+It should count for the majority of the tests we can have. Normally, only test one method each. 
+Example : Whisp, Subscription 
+To run the unit test locally :npm run test:unit:cov
+
+## Integration Tests
```
Comment (MeStrak) [inline docs/testing.md:21]: Please mention the mongo in memory database used for the integration tests

Comment (EL-HARBYLY-Abderrahman) [reply docs/testing.md:21]: I just added it

**Pull request closed without merging**